client/signup/utils.js

```diff
@@ -93,6 +95,25 @@ function getNextStepName( flowName, currentStepName ) {
 	return flow.steps[ indexOf( flow.steps, currentStepName ) + 1 ];
 }
 
+
+function autoFillField( config ) {
```

Comment:
Great idea to move this into its own function in `signup/utils`. A couple super minor notes:

- `autoFillField` is a sort of generic name for what this does. What do you think about `autoFillFieldWithProgressData` or something?
- If we decide to add this functionality to the domains step, the data we'll be using to auto-fill the username will need to be modified, as it will be a domain like `foobarbaz.net`. Splitting this into two functions (one to grab field x from step y and one to add it to the form) would give us the option of modifying the data before it is injected into the form. This also might make naming these functions easier. What do you think?
- _Super_ minor and not necessary to change, but you can include the destructured assignment right in the function signature if you want:

```
function autoFillField( { form, stepName, fieldToTake, fieldToMerge, signupProgress } )
```